libsolidity/analysis/SyntaxChecker.cpp

```diff
@@ -154,3 +176,13 @@ bool SyntaxChecker::visit(PlaceholderStatement const&)
 	return true;
 }
 
+bool SyntaxChecker::visit(MemberAccess const& _memberAccess)
+{
+	if (m_nonPayablePublic && !m_library) {
+		auto const& identifier = dynamic_cast<Identifier const&>(_memberAccess.expression());
```

Comment:
I am not sure this is the right way to perform the cast, but I don't have enough C++ knowledge to fix it.